### PR TITLE
 Removed unused RecoParticleFlow/PFClusterShapeProducer package

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

This PR is in the same spirit as https://github.com/cms-sw/cmssw/pull/26184.

This package was not used anywhere else in CMSSW and the plugin it provided was also not used.

Please, if you know what this code was used for of if it's still necessary, I am happy to learn about it!

#### PR validation:

CMSSW still compiles and local matrix tests pass.

### CODE REVIEW

**Comment by slava77** [comment]
@cbernet @pjanot 
@hatakeyamak 

are there any still possibly planned use cases for RecoParticleFlow/PFClusterShapeProducer?
I checked back to the last times these packages were moved around in CMSSW_3_1_0 and git grep there returns no use cases of the ```pfClusterShapes```.

**Comment by hatakeyamak** [comment]
I will try to have a closer look, but it will be nice to hear from @cbernet and @pjanot

**Comment by perrotta** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

File: Utilities/ReleaseScripts/scripts/git-publish [hunk 1]
```diff
@@ -257,7 +257,6 @@ CMSSW_CATEGORIES={"Operations": ["Configuration/Applications",
   "RecoMuon/TrackingTools", "RecoMuon/TransientTrackingRecHit",
   "RecoParticleFlow/Benchmark", "RecoParticleFlow/Configuration",
   "RecoParticleFlow/PFBlockProducer", "RecoParticleFlow/PFClusterProducer",
-  "RecoParticleFlow/PFClusterShapeProducer", "RecoParticleFlow/PFClusterTools",
```
Comment (Dr15Jones) [inline Utilities/ReleaseScripts/scripts/git-publish]: `RecoParticleFlow/PFClusterTools` should not have been deleted

**Comment by Dr15Jones** [comment]
-1

File: Utilities/ReleaseScripts/scripts/git-publish (hunk 1 above)
Comment (guitargeek) [reply Utilities/ReleaseScripts/scripts/git-publish]: Thanks!

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by slava77** [comment]
The tests passed and there were no objections yet to the removal.
My plan is to leave this open for comments for a few more days and then signoff by the end of the week if there will be no objections.

**Comment by hatakeyamak** [comment]
I checked:
https://github.com/hatakeyamak/PFCalibration
which is a fork of:
https://github.com/spandeyehep/PFCalibration
and I don't see any reference to PFClusterShapeProducer, so I think it's safe to remove it.

The use of shape information is something I think we should be exploring in future, though, but I believe it can be done without this package.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/26186	b5c0b54
- code changes are in line with the PR description and the follow up review
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**